src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs

```diff
@@ -0,0 +1,553 @@
+            // allow the test code to force a transform, despite the fact that the provided version doesn't call for it.
+            string version = forceUpdate ? "2.0.0" : (string)sarifLog["version"];
+
+            switch (version)
+            {
+                // This case actually handles all SARIF v2 prerelease before we 
+                // started properly updating the schema. 
+                case "2.0.0":
+                {
+                    modifiedLog |= ApplyCoreTransformations(sarifLog);
+                    break;
+                }
+
+                case "2.0.0-csd.2.beta.2018-10-10":
+                {
+                    modifiedLog |= ApplyCoreTransformations(sarifLog);
+                    modifiedLog |= ApplyChangesSinceTechnicalCommitteeMeeting25();
+                    break;
+                }
+
+                default:
+                    break;
+            }
+
+            string text = modifiedLog ? sarifLog.ToString(formatting) : prereleaseSarifLog;
+
+            return text;
+        }
+        private static bool ApplyCoreTransformations(JObject sarifLog)
+        {
+            bool modifiedLog = UpdateSarifLogVersion(sarifLog); 
+
+            var runs = (JArray)sarifLog["runs"];
+
+            if (runs != null)
+            {
+                // Move result.ruleMessageId: https://github.com/oasis-tcs/sarif-spec/issues/216
+                // Update name of threadflowLocation.executionTimeUtc: https://github.com/oasis-tcs/sarif-spec/issues/242
+                modifiedLog |= UpdateRunResults(runs);
+
+                // invocation.workingDirectory now a file location: https://github.com/oasis-tcs/sarif-spec/issues/222
+                // Update names of invocation.startTimeUtc and endTimeUtc: https://github.com/oasis-tcs/sarif-spec/issues/242
+                // Convert exception.message to message object: https://github.com/oasis-tcs/sarif-spec/issues/240
+                modifiedLog |= UpdateRunInvocations(runs);
+
+                // run.originalUriBaseIds values now file locations: https://github.com/oasis-tcs/sarif-spec/issues/234
+                modifiedLog |= UpdateRunOriginalUriBaseIds(runs);
+
+                // Convert file.hashes to dictionary: https://github.com/oasis-tcs/sarif-spec/issues/240
+                // Update name of file.lastModifiedTimeUtc: https://github.com/oasis-tcs/sarif-spec/issues/242
+                modifiedLog |= UpdateRunFiles(runs);
+
+                // Update names of notification.startTimeUtc and endTimeUtc: https://github.com/oasis-tcs/sarif-spec/issues/242
+                modifiedLog |= UpdateRunNotifications(runs);
+
+                // Update name to versionControlDetails.repositoryUri: https://github.com/oasis-tcs/sarif-spec/issues/244
+                modifiedLog |= UpdateVersionControlProvenance(runs);
+
+                modifiedLog |= UpdateRunAutomationDetails(runs);
```

Comment:
>Update [](start = 31, length = 6)

Although it's a naming inconsistency with the other methods, a more accurate description of what this method does is `IntroduceRunAutomationDetails`.

Reply:
Well, I meant the name to imply that we were introducing the _object_ `RunAutomationDetails`, not that we were introducing the details themselves -- as you say, they already existed -- but ok.

---
In reply to: [226844923](https://github.com/Microsoft/sarif-sdk/pull/1114#discussion_r226844923) [](ancestors = 226844923,226762406)